### PR TITLE
Issue #74 Set up builds on https://jenkins.eclipse.org/wildwebdeveloper/

### PR DESCRIPTION
**Pull request by jabby**
Signed-off-by: Gautier de Saint Martin Lacaze <gautier.desaintmartinlacaze@gmail.com>

### CODE REVIEW

**Comment by mickaelistria** [comment]
We can see on https://jenkins.eclipse.org/wildwebdeveloper/job/Wildwebdeveloper/view/change-requests/job/PR-77/ that the connection is happening. Good.
I've opened https://bugs.eclipse.org/bugs/show_bug.cgi?id=539780 to get hints about configuring CI or GitHub so that Jenkins votes as a reviewer.
About the error with vncserver missing, I think it would make sense to try using a GitHub image as agent. You may like to try something like https://github.com/eclipse/aCute/pull/148/files to have a clear and controlled defition of the build+test environment that CI should use.

**Comment by jabby** [comment]
We can use docker ? Great news.
I do that this afternoon.

**Comment by mickaelistria** [comment]
I was too eager, actually, we cannot fully use it: 
https://bugs.eclipse.org/bugs/show_bug.cgi?id=539114#c4 > "you can only run jobs in Docker containers, but you can't build Docker images"
https://bugs.eclipse.org/bugs/show_bug.cgi?id=539114#c6 > "you will need to use kubernetes agent [...] Images are pulled from DockerHub."

So at the moment, it's restricted to DockerHub images. If you happen to know a good DockerHub image to use, then we can move forward; otherwise it involves publishing a Docker image on DockerHub...

**Comment by jabby** [comment]
If I read correctly we can't build our own docker image like you did in https://github.com/eclipse/aCute/pull/148 ?

**Comment by mickaelistria** [comment]
Right, a https://github.com/eclipse/aCute/pull/148 is still a failing PR at the moment (so I didn't really did it, I just tried unsuccessfully).

**Comment by mickaelistria** [comment]
As an alternative to make it work, you can open a ticket asking for installation of Xvnc and its Jenkins plugin on the Jenkins instance.

**Comment by mickaelistria** [comment]
Another thing is at the moment to make the Jenkinsfile skip the Xvnc start and the test (adding the -DskipTests) so it will at least proceed the compilation and packaging.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		stage('Prepare') {
+			steps {
+				git url: 'https://github.com/eclipse/wildwebdeveloper.git'
+				cleanWs()
+				checkout scm
+			}
+		}
+		stage('Build') {
+			steps {
+				wrap([$class: 'Xvnc', useXauthority: true]) {
```
Comment (mickaelistria) [inline Jenkinsfile]: Temporarily, let's remove/comment this wrapper until we have Xvnc installed on the machine or can run it in a container.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+					withEnv([]) { // Which environment variable we should use for node ?
+						withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', mavenLocalRepo: '.repository') {
+							sh 'mvn clean verify -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true -PpackAndSign'
```
Comment (mickaelistria) [inline Jenkinsfile]: If we don't have Xvnc, let's pass the `-DskipTests` flag to not run tests at the moment.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+						}
+					}
+				}
+			}
+			post {
+				success {
+					junit '*/target/surefire-reports/TEST-*.xml'
```
Comment (mickaelistria) [inline Jenkinsfile]: Since we don't run tests, let's comment it out at the moment

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+			}
+		}
+		stage('Deploy') {
+			when {
+				branch 'master'
+				// TODO deploy all branch from Eclipse.org Git repo
+			}
+			steps {
+				// TODO compute the target URL (snapshots) according to branch name (0.5-snapshots...)
+				sh 'rm -rf /home/data/httpd/download.eclipse.org/wildwebdeveloper/snapshots'
```
Comment (mickaelistria) [inline Jenkinsfile]: I'm not sure about those path on the new CI instance. But let's just keep them at the moment and fix them later if it happens they're wrong.

**Comment by mickaelistria** [comment]
I've opened https://bugs.eclipse.org/bugs/show_bug.cgi?id=539955 to request installation of Xvnc.

**Comment by mickaelistria** [comment]
It seems too much of a PITA to run UI tests on jenkins.eclipse.org these days.
So I suggest we
1. keep a Jenkinsfile for build/sign/deploy on jenkins.eclipse.org, but we skip tests
2. enable Travis and have a .travis.yml file tailored for testing.

What do you think?

**Comment by jabby** [comment]
I'm ok for the travis ci for tests purpose. 
Just to be sure the main problem here for tests is that we need a docker image with a X server, maven and npm? That's it?

**Comment by mickaelistria** [comment]
Right. But first we need some tests ;) I noticed yesterday we don't have any.
And indeed, that's the Docker image we need to be published. That's what I think is the annoying step: we need to deploy the Dockerfile as an image somewhere (like on dockerhub) just for that sake of our build...

**Pull request merged**